### ISSUE TITLE
Do we need a 'test' set for day-to-day ML research?

### ISSUE THREAD

**Issue by JackKelly** [openclimatefix/nowcasting_dataset#141]
I'm probably doing it wrong, but in my day-to-day ML research, I tend to only use a 'train' and a 'validation' set.

We'd definitely need a hidden 'test' set for something like a public ML competition.

But do we need to split our precious dataset three ways (train / validation / test) for day-to-day ML research?  I'm honestly not sure what best practice is here!  (Which I'm a little embarrassed to admit!)  Should we _always_ have a held-out test set whereby we stop training when the error on the validation set starts increasing and then, after training has finished, we test again on the held-out test set?

(although, IIRC, recent nowcasting papers seem to train for a pre-defined number of epochs, rather than stopping when the validation error starts increasing).

Related issues: #136 #140 #131
